simulation_study/plot-case-study-denisties.R

```diff
@@ -0,0 +1,180 @@
+###########################
+# Function definitions ----
+###########################
+
+source("helper_functions.R") # for GPD density and quantiles
+
+density_discontinuous <- function(x, uniform_lower, uniform_upper, proportion_uniform, gpd_scale, gpd_shape){
+
+  is_undefined <- which(x < uniform_lower)
+  is_uniform <- which((x > uniform_lower) & (x < uniform_upper))
+  is_gpd <- which(x > uniform_upper)
+
+  density <- rep(NA_real_, length(x))
+  density[is_undefined] <- 0
+  density[is_uniform] <- proportion_uniform / (uniform_upper - uniform_lower)
+  density[is_gpd] <- dgpd(x[is_gpd], gpd_shape, gpd_scale, mu = uniform_upper)
+  density[is_gpd] <- density[is_gpd] * (1 - proportion_uniform)
+  return(density)
+}
+
+density_continuous <- function(x, alpha = 1, beta = 0.5){
+  gpd_scale = 0.5
+  gpd_shape = 0.1
+  
+  prop_to_density <- function(x, a = alpha, b = beta){
+    dgpd(x, shape = 0.1, scale = 0.5) * pbeta(x, a, b)
+  } 
+
+  # Area correction to get valid pdf
+    AUC_0_to_1 <- integrate(prop_to_density, 0, 1)$value
+    AUC_1_to_inf <- 1 - pgpd(q = 1,shape = gpd_shape, scale = gpd_scale)
+    total_AUC <- AUC_0_to_1 + AUC_1_to_inf
+
+  ## Correct density values to ensure unit integrabile
+    area_correction_factor <- 1 / total_AUC
+    density <- prop_to_density(x) * area_correction_factor
+  
+  return(density)
+}
+
+density_continuous_2 <- function(x, alpha = 1, beta = 0.5){
+
+  is_undefined <- which(x < 0)
+  is_bgpd <- which((x > 0) & (x < 1))
+  is_gpd <- which(x > 1)
+
+  density <- rep(NA_real_, length(x))
+  density[is_undefined] <- 0
+  density[is_bgpd] <- dgpd(x[is_bgpd], shape = 0.1, scale = 0.5) *
+    pbeta(x[is_bgpd], shape1 = alpha, shape2 = beta)
+
+  ## Calculate area under the curve between 0 and 1
+  n_eval_points = 10001
+  eval_points <- ppoints(n_eval_points)
+  eval_values <- dgpd(eval_points, shape = 0.1, scale = 0.5) *
+    pbeta(eval_points, shape1 = alpha, shape2 = beta)
+
+  q <- mean(c(eval_values[-1], eval_values[-n_eval_points]))  # AUC from 0 to 1
```

Comment:
Not sure I understand this step. Why is the eval_values vector repeated within the mean calculation? And why is the first and last element removed?

Reply:
This was Reimann sum (i.e. midpoint heights * interval width). Have swapped for `integrate()` for clarity.